website/docs/sources.md

```diff
@@ -0,0 +1,289 @@
+
+```yaml
+sources:
+  someSource:
+    context: {}
+```
+
+Note the empty brackets.
+This is an unfortunate syntax requirement to not have `context` considered as a nil value.
+This is the equivelent of the following:
+
+```yaml
+sources:
+  someSource:
+    context:
+      name: "context"
+```
+
+Where `name: "context"`, not to be confused with the source type `context`, is named by convention by the docker CLI.
+Additionally contexts can be passed in from the docker cli: `docker build --build-context <name>=<path>`.
+The `<name>` would be the name to use in your yaml to access it.
+
+This could also be written as below, since the `name: context` is the default and is the main build context passed in by the client:
+
+```yaml
+sources:
+  someSource:
+    context: {}
+```
+
+The `context` source type is considered to be a "directory" source.
+
+### Inline
+
+Inline sources are sources that are defined inline in the Dalec configuration.
+You can only specify one of `file` or `dir` in an inline source.
+Directories cannot be nested in inline sources.
+Filenames must not contain a path separator (`/`).
+
+```yaml
+sources:
+  someInlineFile:
+    inline:
+      # This is the content of the source.
+      file:
+        uid: 0
+        gid: 0
+        permissions: 0644
+        contents: |
+          some content
+          some more content
+
+  someInlineDir:
+    inline:
+      dir:
+        uid: 0
+        gid: 0
+        permissions: 0755
+        files:
+          # This is the content of the source.
+          file1:
+            contents: |
+              some content
+              some more content
+            permissions: 0644
+            uid: 0
+            gid: 0
+          file2:
+            contents: |
+              some content
+              some more content
+            permissions: 0644
+            uid: 0
+            gid: 0
+```
+
+Inline sources with `file` are considered to be "file" sources.
+Inline sources with `dir` are considered to be "directory" sources.
+
+### Docker Image
+
+Docker image sources fetch a docker image from a registry.
+The output of this source is a directory containing the contents of the image.
+
+```yaml
+sources:
+  someDockerImage:
+    image:
+      ref: docker.io/library/alpine:3.14
+```
+
+You can also run commands in the image before fetching the contents.
+This is especially useful for generating content.
+
+```yaml
+sources:
+  someDockerImage:
+    image:
+      ref: docker.io/library/alpine:3.14
+      cmd:
+        dir: / # Default path that command steps are executed in
+        cache_dirs: null # map of cache mounts
+          # /foo: {
+          #   mode: shared # `locked` or `private` are the other options
+          #   key: myCacheKey
+          #   include_distro_key: false # Add the target key from the target being built into the cache key
+          #   include_arch_key: false # add the architecture of the image to run the command in into the cache key
+          # }
```

Comment:
It's a bit hard to see this, also it might just be me but not quite sure what this is used for. So maybe we could add a bit more context on its use or function?
```suggestion
        cache_dirs: # Map of cache mounts. Default value: `null`
          /foo: {
             mode: shared # The other options are `locked` or `private` 
             key: myCacheKey
             include_distro_key: false # Add the target key from the target being built into the cache key
             include_arch_key: false # add the architecture of the image to run the command in into the cache key
          }
```

Reply:
Fixed.